src/openmrs-fetch.ts

```diff
@@ -85,33 +87,45 @@ export function openmrsFetch<T = any>(
        * to help developers understand the problem and debug
        */
 
-      // Attempt to download a response body, if it has one
-      return response.text().then(
-        responseText => {
-          let responseBody = responseText;
-          try {
-            responseBody = JSON.parse(responseText);
-          } catch (err) {
-            // Server didn't respond with json, so just go with the response text string
+      /*
+       *Redirect to given url when redirect on auth failure is enabled
+       */
+      const { redirectAuthFailure } = await getConfig("@openmrs/esm-api");
+      if (
+        redirectAuthFailure.enabled &&
+        redirectAuthFailure.errors.indexOf(response.status) >= 0
+      ) {
+        navigateToUrl(redirectAuthFailure.url);
+        throw new OpenmrsFetchError(url, response, null, requestStacktrace);
```

Comment:
I'm going back and forth on whether we should throw this error. Throwing the error triggers error handling, which is something we might not want to happen when we're already handling the error by causing the navigation to happen. For applications using esm-styleguide, an error triggers a toast/growl that is visually displayed to the user, letting them know that something went wrong. We don't want that in this case.

This might be silly, but one option would be to do this instead:

```js
return new Promise(resolve => {})
```

^ This would prevent the error handling and also not resolve the promise. What are your thoughts? Too weird?

Reply:
I was not aware of error handing showing message by default for all the errors. I also do not see any other way as this function expects a promise to be returned.

Reply:
I tried to do it. But I need to resolve or reject the promise somehow to test it. As we worried that rejecting will show error to user. I resolved it (it may cause problem where the promise it handled in user code).  Yup it is more weird. I unable to think through other workarounds. Please suggest, if we can solve it better.